packages/core/lib/console.js

```diff
@@ -232,9 +267,41 @@ class Console extends EventEmitter {
     const contextVars = {};
 
     abstractions.forEach(abstraction => {
-      contextVars[abstraction.contract_name] = abstraction;
+      // don't overwrite Node's native objects - only load contracts
+      // into the repl context when no conflict exists
+      if (
+        this.replContextNameConflicts[abstraction.contract_name].conflict ===
+        false
+      ) {
+        contextVars[abstraction.contract_name] = abstraction;
+      }
     });
 
+    if (
+      Object.entries(this.replContextNameConflicts).some(
+        ([_key, value]) => value.conflict === true && value.warned === false
+      )
```

Comment:
This condition is determined in the `recordNameConflict` function and can be stored in state, like `alertUserOfNewConflicts` or a better name.